### PR TITLE
feat: implement NotificationBadge component with unread count animation (FAB-252)

### PR DESCRIPTION
**Pull request by vedovelli-ai-team-jr**
Implements Linear FAB-252: Notification Badge Component with Unread Count Animation

## Summary
- Created  component that displays a bell icon with unread count badge
- Reads unread count from the  hook (FAB-179)
- Animated pulse indicator when there are unread notifications
- Count displays as '9+' for counts above 9
- Zero-count state: badge hidden when no unread notifications
- Loading state with visual feedback during data fetch
- Fully accessible with ARIA labels and keyboard support
- Integrated into main navigation header

## Component Features
✅ Bell icon with count bubble overlay
✅ Animated pulse dot on new notifications
✅ '9+' display threshold for large counts
✅ Zero-count hidden state
✅ Loading state with disabled button
✅ ARIA labels and keyboard accessibility
✅ Responsive mobile-friendly styling
✅ TypeScript with proper prop typing

## Files Added
-  - Main component
-  - Comprehensive unit tests
-  - Module exports

## Files Modified
-  - Integrated badge into navbar with onClick handler

## Test Coverage
- Rendering and accessibility attributes
- Unread count display logic
- Badge display threshold (9+)
- Loading and disabled states
- Click handlers and animations
- Edge cases (0 count, boundary values, large counts)

---
Generated by Ana, Junior Developer
Follows Ana's code style: pragmatic, working code first, proper testing

### CODE REVIEW

File: src/components/NotificationBadge/NotificationBadge.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+  return (
+    <button
+      onClick={onClick}
+      disabled={isDisabled}
+      className={`relative p-2 rounded-lg transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-blue-400 focus:ring-offset-2 focus:ring-offset-slate-900 min-h-11 min-w-11 ${
+        isDisabled
+          ? 'opacity-50 cursor-not-allowed'
+          : 'text-slate-400 hover:text-slate-200 hover:bg-slate-800'
+      } ${className}`}
```
Comment (vedovelli-ai-team-sr) [inline src/components/NotificationBadge/NotificationBadge.tsx]: The `aria-pressed` attribute is designed for toggle buttons that maintain state (on/off). Since this button opens the Notification Center rather than toggling a state, `aria-pressed` is semantically incorrect here. Your dynamic `aria-label` already clearly communicates the notification status ("Notifications", "1 unread notification", etc.), so I'd recommend removing this attribute.
